### PR TITLE
feat: add more chat api

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
more chat api, including answer knowledgebase reference, message history, conversation list.
#### Which issue(s) this PR fixes

Fixes #

#### Special notes for your reviewer
doc: https://shimo.im/docs/vVAXMJ6PZ2IwQK3m

### CODE REVIEW

File: apiserver/service/chat.go [hunk 1]
```diff
@@ -57,6 +60,61 @@ func chatHandler() gin.HandlerFunc {
...
+		}
+		c.JSON(http.StatusOK, resp)
+	}
+}
+
+func RegisteryChat(g *gin.RouterGroup, conf config.ServerConfig) {
+	g.POST("", auth.AuthInterceptor(conf.EnableOIDC, oidc.Verifier, "get", "applications"), chatHandler())                         // chat with bot
+	g.POST("/conversations", auth.AuthInterceptor(conf.EnableOIDC, oidc.Verifier, "get", "applications"), conversationHandler())   // current conversations list
+	g.POST("/messages", auth.AuthInterceptor(conf.EnableOIDC, oidc.Verifier, "get", "applications"), historyHandler())             // messages history
+	g.POST("/messages/reference", auth.AuthInterceptor(conf.EnableOIDC, oidc.Verifier, "get", "applications"), referenceHandler()) // messages reference
```
Comment (nkwangleiGIT) [inline apiserver/service/chat.go:119]: reference 应该复数？会不会有对应某个 message 的reference 的需求？那就是类似
/messages
/messages/:message/references

Comment (Abirdcfly) [reply apiserver/service/chat.go:119]: 嗯。也可以。这个请求是个POST请求，mesageID在POST body里。

**Review by 0xff-dev — APPROVED** [review]

File: apiserver/pkg/auth/auth.go [hunk 2]
```diff
@@ -148,17 +153,17 @@ func AuthInterceptor(needAuth bool, oidcVerifier *oidc.IDTokenVerifier, verb, re
 				})
 				return
 			}
+			ctx.Request = ctx.Request.WithContext(context.WithValue(ctx.Request.Context(), UserNameContextKey, userName))
```
Comment (bjwswang) [inline apiserver/pkg/auth/auth.go:156]: Shoule this call `return` immediately ?

File: apiserver/pkg/auth/auth.go [hunk 3]
```diff
@@ -148,17 +153,17 @@ func AuthInterceptor(needAuth bool, oidcVerifier *oidc.IDTokenVerifier, verb, re
...
 		}
 
 		// for graphql query
-		ctx1 := context.WithValue(ctx.Request.Context(), idtokenKey{}, rawToken)
-		ctx.Request = ctx.Request.WithContext(ctx1)
+		ctx.Request = ctx.Request.WithContext(context.WithValue(ctx.Request.Context(), UserNameContextKey, rawToken))
```
Comment (bjwswang) [inline apiserver/pkg/auth/auth.go:160]: Context will always be `rawToken` if you do not return.

Comment (Abirdcfly) [reply apiserver/pkg/auth/auth.go]: oh, there is a typo, I should use idTokenContextKey when store rawToken

**Review by bjwswang — APPROVED** [review]

**Pull request merged**